src/DataCollector/MongoQuerySerializer.php

```diff
@@ -43,19 +43,24 @@ private static function prepareUnserializableData($data): array
      *
      * @return mixed
      */
-    public static function prepareItemData($item)
+    public static function prepareItemData($item, int $depth = 0)
     {
+        // Prevent infinite recursion
+        if ($depth > 1_000) {
+            return null;
+        }
+
         if (\is_scalar($item)) {
             return $item;
         }
 
         if (\is_array($item)) {
-            return self::prepareUnserializableData($item);
+            return self::prepareUnserializableData($item, $depth);
         }
```

Comment:
Why don't we use a maxDepth approach?

Default it to 256 or so, and then decrease it at each iteration `int $maxDepth = 256`

To stop infinite recursion you should add 

```
if ($depth < 0) {
     return null;
}
```

so that also negative `$maxDepth` are correctly managed.

Also, having maxDepth as a parameter cloud make it configurable in the future.

Reply:
Sorry for my very late response, priorities shifted in the meantime. But now the same issue arose so I'm back at it.

I changed the PR to use a $maxDepth, which is definitely better. Thanks for the suggestion.